src/osg/Texture.cpp

```diff
@@ -2310,69 +2720,130 @@ void Texture::applyTexImage2D_load(State& state, GLenum target, const Image* ima
             int width  = inwidth;
             int height = inheight;
 
-            GLenum texStoragesizedInternalFormat = extensions->isTextureStorageEnabled && extensions->isTexStorage2DSupported() && (_borderWidth==0) ? selectSizedInternalFormat(image) : 0;
```

Comment:
New version of lines 2722-2847 (should have both the new changes from OSG 3.6 and the makOsg 3.4 changes. 
```
            GLenum texStoragesizedInternalFormat = extensions->isTextureStorageEnabled && extensions->isTexStorage2DSupported() && (_borderWidth==0) ? selectSizedInternalFormat(image) : 0;

            if (texStoragesizedInternalFormat!=0)
            {
                if (getTextureTarget()==GL_TEXTURE_CUBE_MAP)
                {
                    if (target==GL_TEXTURE_CUBE_MAP_POSITIVE_X)
                    {
                        // only allocate on first face image
                        extensions->glTexStorage2D(GL_TEXTURE_CUBE_MAP, numMipmapLevels, texStoragesizedInternalFormat, width, height);
                    }
                }
                else
                {
                    OsgProfileC("glTexStorage2D(alloc)", tracy::Color::Purple);
                    osg::CVSpan allocSpan(series_rt2, 0, "glTexStorage2D(alloc)");            
                    extensions->glTexStorage2D(target, numMipmapLevels, texStoragesizedInternalFormat, width, height);
                }

                if( !compressed_image )
                {
                   //VRV_PATCH
                   if (target != GL_TEXTURE_2D || !getTextureObjectManager(contextID)->getTextureStreamingActive())
                   {
                      OsgProfileC("sendMipmaps", tracy::Color::Purple);
                      osg::CVSpan UpdateTick(series_rt2, 4, "sendMipmaps");
                      for( GLsizei k = 0 ; k < numMipmapLevels  && (width || height) ;k++)
                      {

                        if (width == 0)
                            width = 1;
                        if (height == 0)
                            height = 1;

                        glTexSubImage2D( target, k,
                            0, 0,
                            width, height,
                            (GLenum)image->getPixelFormat(),
                            (GLenum)image->getDataType(),
                            dataPtr + image->getMipmapOffset(k));

                        width >>= 1;
                        height >>= 1;
                      }
                   }
                   //VRV_PATCH
                   else {
                      // start streaming them
                      textureObject->_currentMipMapToApply = numMipmapLevels-1;
                   }
                }
                else if (extensions->isCompressedTexImage2DSupported())
                {
                   //VRV_PATCH
                   if (target != GL_TEXTURE_2D || !getTextureObjectManager(contextID)->getTextureStreamingActive())
                   {
                      OsgProfileC("sendCompressedMipmaps", tracy::Color::Purple);
                      osg::CVSpan UpdateTick(series_rt2, 4, "sendCompressedMipmaps");

                      GLint blockSize,size;
                      for( GLsizei k = 0 ; k < numMipmapLevels  && (width || height) ;k++)
                      {
                        if (width == 0)
                            width = 1;
                        if (height == 0)
                            height = 1;

                        getCompressedSize(image->getInternalTextureFormat(), width, height, 1, blockSize,size);

                        //state.checkGLErrors("before extensions->glCompressedTexSubImage2D(");

                        extensions->glCompressedTexSubImage2D(target, k,
                            0, 0,
                            width, height,
                            (GLenum)image->getPixelFormat(),
                            size,
                            dataPtr + image->getMipmapOffset(k));

                        //state.checkGLErrors("after extensions->glCompressedTexSubImage2D(");

                        width >>= 1;
                        height >>= 1;
                      }
                   }
                   else {
                      // start streaming them
                      textureObject->_currentMipMapToApply = numMipmapLevels-1;
                   }
                }

```